docs/reference/data_source.rst

```diff
@@ -0,0 +1,52 @@
+.. index::
+    double: Reference; Export / DataSource
+
+Export / DataSource
+===================
+
+When using an admins export feature you might want to modify how dates and times are exported.
+This is done by calling ``setDateTimeFormat`` on the data source iterator.
+
+Here's one way to do it:
+
+1. Decorate the default Sonata\DoctrineORMAdminBundle\Exporter\DataSource with your own and call ``setDateTimeFormat`` there.::
+
+      <?php
+  
+      namespace App\Service\Admin;
+  
+      use Sonata\AdminBundle\Datagrid\ProxyQueryInterface;
+      use Sonata\AdminBundle\Exporter\DataSourceInterface;
+      use Sonata\DoctrineORMAdminBundle\Exporter\DataSource;
+      use Sonata\Exporter\Source\DoctrineORMQuerySourceIterator;
+      use Sonata\Exporter\Source\SourceIteratorInterface;
+  
+      class DecoratingDataSource implements DataSourceInterface
+      {
+        private DataSource $dataSource;
```

Comment:
```suggestion
          private DataSource $dataSource;
```

Please use 4 space indention for php example code

Reply:
Done